File: Prob07/Prob07.java
import java.io.*;
import java.util.*;
import java.text.*;

public class Prob07 {
    public static void main(String[] args) {
        try {
            File f = new File("Prob07.in.txt");
            FileReader fr = new FileReader(f);
            BufferedReader br = new BufferedReader(fr);

            int n = Integer.parseInt(br.readLine());
            for(int i = 0; i < n; i++) {
                String[] input = br.readLine().split(":");
                String[] data = input[1].split(",");
                int[] hits = new int[4];
                int atBats = 0;
                for(int j = 0; j < data.length; j++) {
                    if(data[j].equals("BB") == false) {
                        atBats++;
                        if(data[j].equals("1B")) {
                            hits[0]++;
                        } else if(data[j].equals("2B")) {
                            hits[1]++;
                        } else if(data[j].equals("3B")) {
                            hits[2]++;
                        } else if(data[j].equals("HR")) {
                            hits[3]++;
                        }
                    }
                }

                DecimalFormat formatter = new DecimalFormat("0.000");
                if(atBats == 0) {
                    System.out.println(input[0] + "=" + formatter.format(0));
                } else {
                    double slg = ((double) ((1 * hits[0]) + (2 * hits[1]) + (3 * hits[2]) + (4 * hits[3]))) / atBats;
                    System.out.println(input[0] + "=" + formatter.format(slg));
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: Prob07/Prob07.py
def main():
    with open("Prob07.in.txt") as f:
        myList = [line.rstrip('\n') for line in f]
        n = int(myList[0])
        for i in range(1, n + 1):
            inputs = myList[i].split(":")
            data = inputs[1].split(",")
            hits = [0, 0, 0, 0]
            atBats = 0
            for j in data:
                if j != "BB":
                    atBats += 1
                    if j == "1B":
                        hits[0] += 1
                    elif j == "2B":
                        hits[1] += 1
                    elif j == "3B":
                        hits[2] += 1
                    elif j == "HR":
                        hits[3] += 1

            if atBats == 0:
                print(inputs[0] + "=" + "0.000")
            else:
                slg = float((1 * hits[0]) + (2 * hits[1]) + (3 * hits[2]) + (4 * hits[3])) / atBats
                print(inputs[0] + "=" + "{0:.3f}".format(slg))

if __name__ == "__main__": main()
